src/annotator/integrations/image-text-layer.js

```diff
@@ -0,0 +1,196 @@
+import { ListenerCollection } from '../../shared/listener-collection';
+
+/**
+ * @typedef Rect
+ * @prop {number} left
+ * @prop {number} right
+ * @prop {number} top
+ * @prop {number} bottom
+ */
+
+/**
+ * @param {Rect|null} a
+ * @param {Rect} b
+ */
+function mergeBoxes(a, b) {
+  if (!a) {
+    return b;
+  }
+  return {
+    left: Math.min(a.left, b.left),
+    right: Math.max(a.right, b.right),
+    top: Math.min(a.top, b.top),
+    bottom: Math.max(a.bottom, b.bottom),
+  };
+}
+
+/**
+ * ImageTextLayer maintains a transparent text layer on top of an image
+ * which contains text. This enables the text in the image to be selected
+ * and highlighted.
+ *
+ * This is similar to the one that PDF.js creates for us in our standard PDF
+ * viewer.
+ */
+export class ImageTextLayer {
+  /**
+   * Create a text layer which is displayed on top of `image`.
+   *
+   * @param {Element} image - Rendered image on which to overlay the text layer.
+   *   The text layer will be inserted into the DOM as the next sibling of `image`.
+   * @param {Rect[]} charBoxes - Bounding boxes for characters in the image.
+   *   Coordinates should be in the range [0-1], where 0 is the top/left corner
+   *   of the image and 1 is the bottom/right.
```

Comment:
```suggestion
   *   Coordinates should be in the range [0-1], where origin (0, 0) is the top-left corner.
```

Reply:
To me this reads as saying the bottom right corner is (image width in pixels, image height in pixels), which is not the case. It is (1, 1).